src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -107,10 +107,11 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
                 .orElse(personToEdit.getQuarantineStatus());
         InfectionStatus updatedInfectionStatus = editPersonDescriptor.getInfectionStatus()
                                             .orElse(personToEdit.getInfectionStatus());
+        Index updatedId = editPersonDescriptor.getId().orElse(personToEdit.getId());
         Set<Tag> updatedTags = editPersonDescriptor.getTags().orElse(personToEdit.getTags());
 
         return new Person(updatedName, updatedPhone, updatedEmail, updatedAddress,
-                updatedQuarantineStatus, updatedInfectionStatus, updatedTags);
+                updatedQuarantineStatus, updatedInfectionStatus, updatedId, updatedTags);
```

Comment:
Ids are abstracted away from the user. Users should not be aware of Ids, and thus should not be given the opportunity to change the id of a person. I feel that thus editpersondescriptor should not have an id field and id is automatically retrieved from the person that is being edited.

Reply:
Fixed.